### PR TITLE
(feat) O3-3643: Number of Providers Metrics

### PR DESCRIPTION
**Pull request by lucyjemutai**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

## Summary

https://openmrs.atlassian.net/browse/O3-3643
This PR ensures that when you edit the provider on a saved appointment, the count of Providers booked today does not increases because you are replacing the original provider with the edited one. You’re not adding a second provider.

Additionally, Each provider is counted only once by removing duplicates based on their UUID. 
## Screenshots

[Screencast from 2024-08-29 09-05-14.webm](https://github.com/user-attachments/assets/1aeaf250-f1df-475d-9013-aa28371fb77e)

### CODE REVIEW

File: packages/esm-appointments-app/src/hooks/useClinicalMetrics.ts [hunk 1]
```diff
@@ -45,11 +45,13 @@ export function useAllAppointmentsByDate() {
...
-  const providersArray = data?.data?.filter(({ providers }) => providers !== null) ?? [];
-  const providersCount = uniqBy(
-    providersArray.map(({ providers }) => providers).flat(),
-    (provider) => provider.uuid,
-  ).length;
+  const providersArray = data?.data?.flatMap(({ providers }) => providers ?? []) ?? [];
+
+  const validProviders = providersArray.filter((provider) => provider.response === 'ACCEPTED');
+
+  const uniqueProviders = uniqBy(validProviders, (provider) => provider.uuid);
```
Comment (CynthiaKamau) [inline packages/esm-appointments-app/src/hooks/useClinicalMetrics.ts:52]: Thanks @lucyjemutai , whats the use case for this?

Comment (lucyjemutai) [reply packages/esm-appointments-app/src/hooks/useClinicalMetrics.ts:52]: @CynthiaKamau  When an appointment is edited and the provider is changed, the new provider's response is updated to "ACCEPTED," while the previously assigned provider's status is changed to "CANCELLED." All providers ever associated with that patient are listed, including those with outdated statuses. So, to ensure we only count the active providers for the current day, have filter out those with a status other than 'ACCEPTED'. This way, we accurately display the count of active providers per day. For example, in the payload preview-->
_"name": "John Khmer",
  "providers": [
    { "uuid": "8b97c410-67c5-4e49-9a93-265a23b4dd45", "response": "CANCELLED", "name": "Kelvin" },
    { "uuid": "705f5791-07a7-44b8-932f-a81f3526fc98", "response": "ACCEPTED", "name": "Jane Nurse" },
    { "uuid": "f39e57d8-1185-4199-8567-6f1eeb160f05", "response": "CANCELLED", "name": "Super User" },
    { "uuid": "37b8d5ab-49a5-460d-947d-1a385dfb56f2", "response": "CANCELLED", "name": "Bad Man Josh" }
  ]_

Comment (CynthiaKamau) [reply packages/esm-appointments-app/src/hooks/useClinicalMetrics.ts:52]: Thanks for the clarification

**Review by CynthiaKamau — APPROVED** [review]

**Review by mogoodrich — APPROVED** [review]
Looks great, thanks @lucyjemutai !  Merging this in!

**Pull request merged**